### PR TITLE
Test for new TypedArray iterator detach logic

### PR DESCRIPTION
**Pull request by littledan**
Tests new logic in https://github.com/tc39/ecma262/pull/724

Not ready to merge yet, pending discussion in https://github.com/tc39/ecma262/issues/713

### CODE REVIEW

**Comment by bterlson** [comment]
What's the plan for $DETACHBUFFER? Currently it's just a stub that throws, right?

**Comment by caitp** [comment]
TC39 proposal for ArrayBuffer.detach(), just for fun?

**Comment by bterlson** [comment]
I mean, there's some argument for it - if you want to test some functionality in the presence of detached buffers, it'd be nice to have a standard way to make one.

**Comment by leobalter** [comment]
> I mean, there's some argument for it - if you want to test some functionality in the presence of detached buffers, it'd be nice to have a standard way to make one.

I mentioned this before. I'm also sad ArrayBuffer.transfer is no longer active too.

`$DETACHBUFFER` should be replaced by any non-es-standard way to detach a buffer, if that's not available, skip tests that includes `detachArrayBuffer.js`.

**Comment by littledan** [comment]
Plan? I think it's in pretty good shape--the web platform exposes it with `postMessage`, and multiple harnesses have ways of calling into that same hook. I don't see a problem. What's our plan for the way that the module tests require harness integration?

**Comment by littledan** [comment]
@leobalter I don't think harnesses which don't implement `$DETACHBUFFER` should get a free pass for not having it. Those should count as failures, if we want to have a fair comparison.

**Comment by leobalter** [comment]
@littledan yes. that's what we are already doing, and `$DETACHBUFFER` does it nicely mentioning why it's failing.

Still, that allows anyone to monkey-patch files including `detachArrayBuffer.js` or simply ignore them, we can't control that.

**Comment by bterlson** [comment]
> What's our plan for the way that the module tests require harness integration?

I'm going to teach eshost how to module. Once done it should paper over the differences in running modules in browsers vs. command line hosts. Shouldn't a similar thing be done for $DETACHBUFFER? Eg. $.detachBuffer in eshost maybe?

**Comment by littledan** [comment]
`detachBuffer` in eshost sounds good to me. Do you anticpate anything in the test262 repository changing as a result?

**Comment by bterlson** [comment]
@littledan presumably $DETACHBUFFER would simply call $.detachBuffer (or else tests could be updated to call this directly).

**Comment by littledan** [comment]
Updated the test for $.detachArrayBuffer and minor spec text change.

**Comment by tcare** [comment]
Looks good

**Comment by littledan** [comment]
Well, this isn't quite correct until that spec change is merged. However, I hope it gets merged soon.

**Comment by tcare** [comment]
Alright. Let me know if you want me to undo it.

**Comment by bterlson** [comment]
I pulled in tc39/ecma262#724 so no revert needed :-P
